### PR TITLE
fix: tax calculations not getting updated on applying promo code

### LINKED ISSUE #7871: Tax doesn't not gets updated while using the discount code functionality
**Describe the bug**

**To Reproduce**
1. Create an event with a tax and discount code
2. Go to the event page and enter the discount code and apply. See that it doesn't get updated in the price section

Screenshot from 2021-09-28 19-36-13
3. Now order the ticket. Again you can see the tax doesn't get updated in the price section. Also, total section needs to be updated
Screenshot from 2021-09-28 19-37-27
4. Now buy the ticket. See the same error.
**Expected behavior**

Tax calculation is still buggy we need to fix this and we need to move these caculations to the backend.

### PR DESCRIPTION
**Pull request by pc-beast**
Fixes #7871
Test with https://github.com/fossasia/open-event-server/pull/8258

- Changes in order page remaining
#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by pc-beast** [comment]
image

**Comment by pc-beast** [comment]
@maze-runnar is this the correct approach? We can also set originalTax as it is now getting fetched from backend

**Comment by maze-runnar** [comment]
> @maze-runnar is this the correct approach? We can also set originalTax as it is now getting fetched from backend

So far, good

**Comment by maze-runnar** [comment]
@pc-beast please complete this PR first before any other.

**Comment by pc-beast** [comment]
@maze-runnar please test

**Comment by pc-beast** [comment]
image
image
Works. I wonder what might be missing. :thinking:

**Comment by maze-runnar** [comment]
@pc-beast it's updating tax only I am selecting %percent in discount code. When I am selecting Amount, it's not updating tax. Rest of things are working fine.
Screenshot from 2021-10-08 08-43-43

**Comment by pc-beast** [comment]
Please test it with https://github.com/fossasia/open-event-server/pull/8277

File: app/templates/components/orders/order-summary.hbs [hunk 1]
```diff
@@ -101,40 +101,58 @@
...
+                      {{/if}}
+                      {{#if (and this.event.tax.name (not-eq ticket.type 'free'))}}
                         <span>
                           <small class="ui gray-text tiny aligned right">({{this.event.tax.name}})</small>
                         </span>
                       {{/if}}
                     </p>
                   </td>
-                  {{#if this.data.discountCode}}
+                  {{!-- {{#if this.data.discountCode}}
```
Comment (maze-runnar) [inline app/templates/components/orders/order-summary.hbs:145]: Here too

File: app/templates/components/orders/order-summary.hbs [hunk 2]
```diff
@@ -101,40 +101,58 @@
                     </div>
                   </td>
                   <td>
-                    <Orders::TicketPrice
-                      @ticket={{ticket}}
-                      @currency={{this.eventCurrency}}
-                      @amount={{ticket.price}} />
+                    <div class="{{if ticket.discountedTicketPrice 'strike text'}}">
+                      <Orders::TicketPrice
```
[the remark this answers is not in the document]
Comment (pc-beast) [reply app/templates/components/orders/order-summary.hbs:105]: If the ticket has discount code, then make original price strikethrough

**Comment by maze-runnar** [comment]
> ### Updated SS:
> image
> 
> / image

why tax for 6$ ticket is not updated in screenshot?

**Review by maze-runnar — APPROVED** [review]

**Pull request merged**